src/plugin/plugin.cc

```diff
@@ -768,17 +773,29 @@ void plugin::Zeek_Spicy::Plugin::searchModules(const std::string& paths) {
         if ( trimmed_dir.empty() )
             continue;
 
-        if ( ! hilti::rt::filesystem::is_directory(trimmed_dir) ) {
-            ZEEK_DEBUG(hilti::rt::fmt("Module directory %s does not exist, skipping", trimmed_dir));
+        std::error_code ec;
+        if ( auto is_directory = hilti::rt::filesystem::is_directory(trimmed_dir, ec); ec || ! is_directory ) {
+            ZEEK_DEBUG(hilti::rt::fmt("Module directory %s cannot be read, skipping", trimmed_dir));
             continue;
         }
 
         ZEEK_DEBUG(hilti::rt::fmt("Searching %s for *.hlto", trimmed_dir));
 
-        for ( const auto& e : hilti::rt::filesystem::recursive_directory_iterator(trimmed_dir) ) {
-            if ( e.is_regular_file() && e.path().extension() == ".hlto" )
-                loadModule(e.path());
+        auto it = hilti::rt::filesystem::recursive_directory_iterator(trimmed_dir, ec);
+        if ( ! ec ) {
+            while ( it != hilti::rt::filesystem::recursive_directory_iterator() ) {
+                if ( it->is_regular_file() && it->path().extension() == ".hlto" )
+                    loadModule(it->path());
+
+                if ( it.increment(ec); ec ) {
+                    ZEEK_DEBUG(hilti::rt::fmt("Error iterating over %s, skipping any remaining files: %s", trimmed_dir,
```

Comment:
Is that and below worth a warning or warn once (when errno != EEXIST)? I can picture a debugging session due to non-loaded modules until someone pulls out `strace` to figure out there are actually permission errors.

Reply:
Good point, promoted that one and the one on skipping iteration to warnings.